dicom_utils/dicom_factory.py

```diff
@@ -199,20 +204,24 @@ def __call__(self, seed: Optional[int] = None, **kwargs) -> FileDataset:
 
         # Rows/Columns may have been changed without changing PixelData. This will cause
         # an error when reading `dcm.pixel_array`. Try reading pixel_array and if it fails
-        # update pixels with new data of the correct shape
-        try:
-            dcm.pixel_array
-        except ValueError:
-            arr = self.pixel_array(
-                dcm.Rows,
-                dcm.Columns,
-                dcm.NumberOfFrames,
-                dcm.BitsStored,
-                dcm.BitsAllocated,
-                dcm.PhotometricInterpretation,
-                seed=self.seed,
-            )
-            dcm = set_pixels(dcm, arr, dcm.file_meta.TransferSyntaxUID)
+        # update pixels with new data of the correct shape. We will only perform the pixel update
+        # if requested because it can be slow.
+        if self.pixels if pixels is None else pixels:
```

Comment:
Is this equivalent to `if self.pixels and (pixels is None or pixels)` ?

Reply:
If a value other than `None` is passed for `pixels`, that value should override the value of `self.pixels`. In what you wrote, if `self.pixels == False` and `pixels == True` the if statement would not be executed. I originally had this parenthesized to make it more clear but I think the style checker removed it.

Reply:
Gotcha, that makes sense. Maybe this would be slightly clearer?
```python
pixels = self.pixels if pixels_override is None else pixels_ovverride
if pixels:
    ...
```

Reply:
Looks good, made that change.